### ISSUE TITLE
Adaptive Layers - Minimum and Maximum Height instead of variation

### ISSUE THREAD

**Issue by Slashee-the-Cow** [Ultimaker/Cura#17300]
### Is your feature request related to a problem?

I find it limiting that when I use adaptive layers, I can only set a maximum variation that controls both thicker and thinner layers. I'd like to be able to set a minimum and maximum layer height to match what my printer can do.

If, for example, I'm printing at 0.16mm layers, I know that I can go at most about 0.06mm thinner (0.1mm) but I can go at least 0.12mm thicker (0.28mm). However so I don't go below the minimum I can only have the maximum variation set to 0.06mm, meaning the thickest layers will go is 0.22mm.

I print a lot of things with some curved sections and some big, straight sections, it would save time if I could set the maximum layer thickness to 0.28mm for those sections without sacrificing the thinner layers for the curves.

It also means that if you set the layer height to the minimum or maximum reliable height, you can't use adaptive layers because you don't want it to try going any thinner/thicker.

### Describe the solution you'd like

Instead of an _Adaptive Layers Maximum Variation_ setting, have, for example, _Adaptive Layers Minimum Height_ and _Adaptive Layers Maximum Height_ so that it's possible to take advantage of the full range of layer heights a printer can reliably print, instead of being bounded by the minimum or maximum, whichever is closer to your normal layer height.

### Describe alternatives you've considered

About the only "alternative" I can think of is to set the layer height to the exact middle of the range of heights you can print at. However this takes control away from a desired quality and makes you rely entirely on the adaptive layers algorithm to decide the right layer heights for each section.

### Affected users and/or printers

Any printer can benefit (okay, there's probably a printer or ten out there with a fixed layer height), and while many people don't find the adaptive layers setting because it's in the experimental section (or they unwittingly use it because some quality presets turn it on) there are plenty of people who use it and I doubt I'm the only one of them who would like to have this option.

### Additional information & file uploads

I know that adaptive layers can be a problem because it can make the material flow change significantly, but that can be controlled by the _Adaptive Layers Variation Step Size_ setting, which could be made lower by default. You could also introduce a setting that requires at least <x> layers have to be printed per step, so the flow has more time to adjust.

**Comment by LilBub** [comment]
I agree.  I am noticing on some of my recent slices that adaptive layers (AL) will try to go down as low as 0.05 layers (or lower) with a default resolution of 0.15 and AL set to default.   AL works really great most of the time but as I print a bit finer, it tends to go to unrealistic values.

**Comment by cmp-nct** [comment]
Setting upper and lower bounds is definitely superior to the current "in the middle" approach.
A 3d printer and each filament has a minimum height resolution it can work with and a maximum height where quality degrades or flow collapses.

So adding two settings, to set clear bounds would be a very simple solution, the algorithm needs minimal changes.
Just min() and max() with two settings and you can then choose any other settings, resting in peace that it won't go to 0.4 layer height or stick around 0.01.

In addition there is an issue with adaptive layers when the surface is very small. In that case adaptive layers does not increase print speed by increasing layer height but it causes failures or quality issues because of the heat buildup.
The minimal layer time should also act as an upper bound to adaptive layer height - if min layer time is reached then height can should not be increased further (maybe a checkbox)
